server/src/main/java/org/elasticsearch/index/codec/PerFieldMapperCodec.java

```diff
@@ -64,10 +66,16 @@ private PostingsFormat internalGetPostingsFormatForField(String field) {
         return super.getPostingsFormatForField(field);
     }
 
-    private boolean useBloomFilter(String field) {
-        return IdFieldMapper.NAME.equals(field)
-            && mapperService.mappingLookup().isDataStreamTimestampFieldEnabled() == false
-            && IndexSettings.BLOOM_FILTER_ID_FIELD_ENABLED_SETTING.get(mapperService.getIndexSettings().getSettings());
+    boolean useBloomFilter(String field) {
+        IndexSettings indexSettings = mapperService.getIndexSettings();
+        if (mapperService.mappingLookup().isDataStreamTimestampFieldEnabled()) {
+            return indexSettings.getIndexVersionCreated().onOrAfter(Version.V_8_7_0)
```

Comment:
We don't need the onOrAtfter check: PerFieldPostingsFormat records what fields use what format on disk, so old indices would keep using the normal postings format automatically.

Reply:
Can you also add a comment about why we're treating TSDB data streams differently?